enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/ParameterSettings.tsx

```diff
@@ -0,0 +1,114 @@
+import { useDebouncedCallback } from "@mantine/hooks";
+import { useCallback, useMemo } from "react";
+import { P, match } from "ts-pattern";
+import { t } from "ttag";
+
+import { Stack, Text, TextInput } from "metabase/ui";
+
+import { useSdkIframeEmbedSetupContext } from "../context";
+import { useHideParameter } from "../hooks";
+
+import { ParameterVisibilityToggle } from "./ParameterVisibilityToggle";
+
+export const ParameterSettings = () => {
+  const {
+    experience,
+    settings,
+    updateSettings,
+    availableParameters,
+    isLoadingParameters,
+  } = useSdkIframeEmbedSetupContext();
+
+  const { isParameterHidden, toggleParameterVisibility } = useHideParameter();
+
+  const isQuestionOrDashboardEmbed =
+    !!settings.questionId || !!settings.dashboardId;
+
+  const updateInitialParameterValue = useDebouncedCallback(
+    useCallback(
+      (paramId: string, value: string) => {
+        if (settings.dashboardId) {
+          updateSettings({
+            initialParameters: {
+              ...settings.initialParameters,
+              [paramId]: value,
+            },
+          });
+        } else if (settings.questionId) {
+          updateSettings({
+            initialSqlParameters: {
+              ...settings.initialSqlParameters,
+              [paramId]: value,
+            },
+          });
+        }
+      },
+      [settings, updateSettings],
+    ),
+    500,
```

Comment:
This feels magic, we should probably use some predefined debounce value if exists, or create a new one. As this could easily get out of date and hard to make it consistent with the rest of the debounce timing.

Reply:
Good callout, fixed in https://github.com/metabase/metabase/pull/60233/commits/ec13e9f6c365b2cd66378659437923285c39f803. I added some explanations in the constant too why the debounce is needed.